cpp/src/arrow/record_batch.h

```diff
@@ -310,7 +310,20 @@ class ARROW_EXPORT RecordBatchReader {
   /// \brief Read the next record batch in the stream. Return null for batch
   /// when reaching end of stream
   ///
-  /// \param[out] batch the next loaded batch, null at end of stream
+  /// example:
+  /// ```
+  /// std::shared_ptr<RecordBatch> batch;
+  /// while (true) {
+  ///   ARROW_RETURN_NOT_OK(reader->ReadNext(&batch));
```

Comment:
Should we reduce a local variable scope as much as possible?

```suggestion
  /// while (true) {
  ///   std::shared_ptr<RecordBatch> batch;
  ///   ARROW_RETURN_NOT_OK(reader->ReadNext(&batch));
```